site-graph/cache/cachemap/map.go

```diff
@@ -0,0 +1,36 @@
+package cachemap
+
+import (
+	"sync"
+
+	"github.com/adamplansky/go-bridge-mentoring/site-graph/cache"
+)
+
+// cmap implements cache interface and use golang map as underlying
+// data struct for storing objects
+
+type cachemap struct {
+	cache map[string]interface{}
+	// https://stackoverflow.com/questions/53427824/what-is-the-difference-between-rlock-and-lock-in-golang
+	mu sync.RWMutex
+}
+
+func New() cache.Cache {
+	return &cachemap{
+		cache: make(map[string]interface{}),
+		mu:    sync.RWMutex{},
+	}
+}
+
+func (c *cachemap) Add(key string, value interface{}) {
+	c.mu.Lock()
+	c.cache[key] = value
+	c.mu.Unlock()
+}
+
+func (c *cachemap) Get(key string) (value interface{}, ok bool) {
+	c.mu.RLock()
+	value, ok = c.cache[key]
+	c.mu.RUnlock()
+	return
```

Comment:
don't do like that event if it is allowed

named output variables are ok but "naked return" is not readable